test/smoketest_artman.py

```diff
@@ -137,4 +131,16 @@ def _setup_logger(log_file):
 if __name__ == '__main__':
     flags = parse_args(*sys.argv[1:])
 
-    run_smoke_test(os.path.abspath(flags.root_dir), os.path.abspath(flags.log))
+    try:
```

Comment:
This can be simplified:

```
try:
  run_smoke_test(...)
except:
  sys.exit(...)
finally:
  with open(...) as logs:
    print(logs.read())

```

Or:

```
def run_smoke_test(...):
  ...
  if failure:
    logger.error('Smoke test failed')

...

if __name__ == '__main__':
  try:
    run_smoke_test(...)
  finally:
    with open(...) as logs:
      print(logs.read())
```

Reply:
The smoketest would need to exit with non-zero status code when it fails.

Reply:
For nonzero exit status:

```
def run_smoke_test(...):
  ...
  assert(not failure)

...

if __name__ == '__main__':
  try:
    run_smoke_test(...)
  finally:
    with open(...) as logs:
      print(logs.read())
```